jme3-materialeditor/src/com/jme3/gde/materialdefinition/EditableMatDefFile.java

```diff
@@ -1,408 +1,400 @@
-        }
-        if (sta.getContents() != null) {
-            for (Statement statement : sta.getContents()) {
-                registerListener(statement);
-            }
-        }
-    }
-
-    public void buildOverview(ExplorerManager mgr) {
-        if (materialDef != null) {
-            mgr.setRootContext(new MatDefNode(lookup));
-
-        } else {
-            mgr.setRootContext(Node.EMPTY);
-        }
-    }
-
-    public String getShaderCode(String version, Shader.ShaderType type) {
-        try {
-            material.selectTechnique(currentTechnique.getName(), SceneApplication.getApplication().getRenderManager());
-            Shader s;
-            StringBuilder sb = new StringBuilder();
-            TechniqueDef def = material.getActiveTechnique().getDef();
-            sb.append(def.getShaderPrologue());
-            material.getActiveTechnique().getDynamicDefines().generateSource(sb, Arrays.asList(def.getDefineNames()), Arrays.asList(def.getDefineTypes()));
-            
-            if (version.equals(GLSL100)) {
-                glsl100.initialize(material.getActiveTechnique().getDef());
-                s = glsl100.generateShader(sb.toString());
-            } else {
-                glsl150.initialize(material.getActiveTechnique().getDef());
-                s = glsl150.generateShader(sb.toString());
-            }
-            for (Shader.ShaderSource source : s.getSources()) {
-                if (source.getType() == type) {
-                    return source.getSource();
-                }
-            }
-            return "";
-        } catch (Exception e) {
-            Exceptions.printStackTrace(e);
-            return "Error generating shader: " + e.getMessage();
-        }
-    }
-
-    public TechniqueBlock getCurrentTechnique() {
-        return currentTechnique;
-    }
-    
-    public void setCurrentTechnique(TechniqueBlock tech){
-        this.currentTechnique = tech;
-    }
-
-    public MatDefBlock getMatDefStructure() {
-        return matDefStructure;
-    }
-    private final MatStructChangeListener changeListener = new MatStructChangeListener();
-    J3MLoader loader = new J3MLoader();
-
-    private void updateLookupWithMaterialData(MatDefDataObject obj) {
-        obj.getLookupContents().add(materialDef);
-        material = new Material(materialDef);
-
-        try {
-            //material.selectTechnique("Default", SceneApplication.getApplication().getRenderManager());
-            if (matToRemove != null) {
-                for (MatParam matParam : matToRemove.getParams()) {
-                    try {
-                        material.setParam(matParam.getName(), matParam.getVarType(), matParam.getValue());
-                    } catch (IllegalArgumentException ie) {
-                        matToRemove.clearParam(matParam.getName());
-                    }
-
-                }

```

Comment:
This is what caused the mat previews not to update when changing parameters. They got overwritten when the material was reloaded.